clang/lib/Sema/SemaBounds.cpp

```diff
@@ -2222,17 +2224,41 @@ namespace {
       const FunctionType *FuncTy = PointeeType->getAs<FunctionType>();
       assert(FuncTy);
       const FunctionProtoType *FuncProtoTy = FuncTy->getAs<FunctionProtoType>();
-      if (!FuncProtoTy)
+
+      // If the callee and arguments will not be traversed as part of the
+      // checking below, traverse them here.  This prevents TraverseStmt
+      // from needing to traverse the children of call expressions.
+      if (!FuncProtoTy) {
+        TraverseChildren(E, CSS, Facts, SE);
         return ResultBounds;
-      if (!FuncProtoTy->hasParamAnnots())
+      }
+      if (!FuncProtoTy->hasParamAnnots()) {
+        TraverseChildren(E, CSS, Facts, SE);
+        return ResultBounds;
+      }
+      if (SE == SideEffects::Disabled) {
+        TraverseChildren(E, CSS, Facts, SE);
```

Comment:
If there are no side-effects, do you still need to call TraverseChildren here?   What does it do?